extensions/podman/packages/extension/src/podman-machine-stream.spec.ts

```diff
@@ -0,0 +1,217 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { Client } from 'ssh2';
+import { beforeEach, describe, expect, test, vi } from 'vitest';
+
+import type { MachineInfo } from './extension';
+import { ProviderConnectionShellAccessImpl } from './podman-machine-stream';
+
+const onMock = vi.fn();
+const onStreamMock = vi.fn();
+const streamMock = {
+  on: onStreamMock,
+  write: vi.fn(),
+};
+
+beforeEach(() => {
+  vi.clearAllMocks();
```

Comment:
```suggestion
  vi.resetAllMocks();
```

Reply:
For seme reason the resetAllMocks does not work (all of mine tests fails), but it works with clear, any ideas?

Reply:
probably due to
```
 EventEmitter: vi.fn().mockImplementation(() => ({
      fire: vi.fn(),
      dispose: vi.fn(),
    })),
```

Reply:
(all the mock implementations you're doing in mocking the modules)